### ISSUE TITLE
ostream argument in integrate_1d

### ISSUE THREAD

**Issue by bbbales2** [stan-dev/math#933]
#### Summary:
@bob-carpenter 

Should the ostream argument that gets passed through integrate_1d to the thing being integrated be a pointer instead of a reference?

Currently we have it as a reference, but I just did a stanc on an ode model and get the signature:
```
...      const std::vector<int>& x_i, std::ostream* pstream__) {
```

which seems to indicate it should be a pointer.

If it should be a pointer, leave the issue open and I'll fix it. If not, close it and we're good.

#### Current Version:
develop (0b0e0a4788774157fdefe77952428394faa15260)

**Comment by bob-carpenter** [comment]
`*x` will convert pointer `x` to something that can be passed by reference.  So that's not a problem.  In fact, we should be moving to eliminate pointers everywhere, so I'd say it's a positive.  

The bigger problem is that I can't compile a double-only call to `integrate_1d`.  I'm going to take that up with you via email.

**Comment by andrjohns** [comment]
Closed via #936

**Comment by seantalts** [comment]
I'm going to reopen this - looks like integrate_1d is still a reference. I'm happy to move to references but I strongly suggest we do such code migrations in a single refactoring swoop rather than piecemeal for API-level stuff like this.
